libcodechecker/server/product_db_access_handler.py

```diff
@@ -0,0 +1,300 @@
+                             'dbport': dbc.port,
+                             'dbusername': dbuser,
+                             'dbpassword': dbpass,
+                             'dbname': dbc.database}
+        elif dbc.engine == 'sqlite':
+            conn_str_args = {'postgresql': False,
+                             'sqlite': dbc.database}
+        else:
+            msg = "Database engine '{0}' unknown!".format(dbc.engine)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.GENERAL,
+                                              msg)
+
+        conn_str = SQLServer.connection_string_from_cmdline_args(conn_str_args)
+
+        # Create the product's entity in the database.
+        try:
+            orm_prod = Product(
+                endpoint=product.endpoint,
+                conn_str=conn_str,
+                name=displayed_name,
+                description=description)
+
+            LOG.debug("Attempting database connection to new product...")
+
+            # Connect and create the database schema.
+            self.__server.add_product(orm_prod)
+            LOG.debug("Product database successfully connected to.")
+
+            connection_wrapper = self.__server.get_product(product.endpoint)
+            if connection_wrapper.last_connection_failure:
+                msg = "The configured connection for '/{0}' failed: {1}" \
+                    .format(product.endpoint,
+                            connection_wrapper.last_connection_failure)
+                LOG.error(msg)
+
+                self.__server.remove_product(product.endpoint)
+
+                raise shared.ttypes.RequestFailed(
+                    shared.ttypes.ErrorCode.IOERROR, msg)
+
+            self.__session.add(orm_prod)
+            self.__session.commit()
+            LOG.debug("Product configuration added to database successfully.")
+
+            # The orm_prod object above is not bound to the database as it
+            # was just created. We use the actual database-backed configuration
+            # entry to handle connections, so a "reconnect" is issued here.
+            self.__server.remove_product(product.endpoint)
+
+            orm_prod = self.__session.query(Product) \
+                .filter(Product.endpoint == product.endpoint).one()
+            self.__server.add_product(orm_prod)
+
+            LOG.debug("Product database connected and ready to serve.")
+            return True
+
+        except sqlalchemy.exc.SQLAlchemyError as alchemy_ex:
+            msg = str(alchemy_ex)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.DATABASE,
+                                              msg)
+
+    @timeit
+    def removeProduct(self, product_id):
+        """
+        Disconnect the product specified by the ID from the server.
+        """
+
+        try:
+            product = self.__session.query(Product) \
```

Comment:
Isn't attempting to remove a non-existent thing a failure? I agree with an early-"return", but in form of an early `raise`.